tenacity/__init__.py

```diff
@@ -412,6 +414,8 @@ def __init__(self, retry_object, fn, args, kwargs):
 
         #: The number of the current attempt
         self.attempt_number = 1
+        #: Tracks exceptions raised by the method, by type
+        self.exception_counts = Counter()
```

Comment:
There should not be any need to store the number of exceptions here. The `stop_after_exception_count` class can store them AFAIU.

Reply:
You're right, I thought it would keep state between calls, updated